modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/options/DefaultOptions.java

```diff
@@ -0,0 +1,309 @@
+package org.openapitools.codegen.languages.options;
+
+import com.google.common.base.Strings;
+import org.openapitools.codegen.CodegenConfig;
+import org.openapitools.codegen.CodegenConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+public abstract class DefaultOptions implements Options {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultOptions.class);
+
+    protected Set<String> reservedWords;
+    protected Map<String, String> typeMapping;
+    protected Map<String, String> nullableTypeMapping;
+    protected Map<String, String> importMapping;
+    protected Set<String> languageSpecificPrimitives;
+    protected Map<String, String> instantiationTypes;
+    protected Set<String> collectionTypes;
+    protected Set<String> mapTypes;
+    protected Set<String> defaultIncludes;
+    protected Map<String, Object> primitiveDefaults;
+    protected Set<String> languageGenericTypes;
+    protected Map<String, String> specialCharReplacements;
+    protected boolean allowUnicodeIdentifiers;
+    protected boolean supportsInheritance;
+    protected boolean supportsMixins;
+
+    public DefaultOptions() {
+        defaultIncludes = new HashSet<String>(
```

Comment:
I see. I guess the problem is even if we use the overall Java language in the generator we might not want to use ArrayList in a given Java framework so adding the `instantiationTypes.put("array", "ArrayList");` in the JavaOptions doesn't really make sense. 

I saw that: in plenty of places during the extraction, there was mappings overriden by a given framework (plenty of `typeMapping().put(` or `importMapping().put(` which are just a sign that something's wrong (modifying a collection accessed through a getter).

I've put some defaults in the DefaultOptions constructor because they were in the DefaultCodegen constructor, but I guess from what you're saying here they shouldn't be there either ;-) (`importMapping.put("BigDecimal", "java.math.BigDecimal");` is done for all *Codegen for all generators because they all extend DefaultCodegen)